src/python.rs

```diff
@@ -67,9 +71,15 @@ impl<'j> PythonParser<'j> {
                 self.parser.consume_false()?;
                 Ok(false.to_object(py))
             }
-            Peek::Null => {
-                self.parser.consume_null()?;
-                Ok(py.None())
+            Peek::Minus | Peek::Infinity | Peek::NaN => {
+                let n = self
+                    .parser
+                    .consume_number::<NumberAny>(peek.into_inner(), self.allow_inf_nan)?;
```

Comment:
Maybe one solution is to remove `peek.into_inner()` and replace it with `peek.into_number()` which errors when it's not a number? This would probably keep most of the same performance properties, but force you to handle all invalid cases earlier as part of the wildcard / numerical branch.